### PR TITLE
python3Package.mergedb: init at 0.1.1

### PR DESCRIPTION
**Pull request by graysonhead**
###### Description of changes
Python3 module and associated CLI utility to perform deep merging of YAML/JSON files for automation tools. Homepage: https://github.com/graysonhead/mergedb

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4656,6 +4656,12 @@
     githubId = 4656860;
     name = "Gaute Ravndal";
   };
+  graysonhead = {
```
Comment (fabaff) [inline maintainers/maintainer-list.nix:4659]: Could you please move this change to a separate commit in this PR?

File: pkgs/development/python-modules/mergedb/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{ lib, pkgs, fetchPypi, setuptools, pyyaml, colorama, jinja2, setuptools-scm, git, python3Packages }:
```
Comment (fabaff) [inline pkgs/development/python-modules/mergedb/default.nix:1]: One input per line makes the diffs in the future easier to read.

```suggestion
{ lib, 
, buildPythonPackage
, colorama
, fetchPypi
, jinja2
, pytestCheckHook
, pythonOlder
, pyyaml
, setuptools-scm
}:
```

File: pkgs/development/python-modules/mergedb/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+let
+  pname = "mergedb";
+  version = "0.1.1";
+in
+python3Packages.buildPythonPackage {
+  inherit pname version;
```
Comment (fabaff) [inline pkgs/development/python-modules/mergedb/default.nix:8]: I suggest to disable the module on all releases which are EOL.

```suggestion
buildPythonPackage rec {
  pname = "mergedb";
  version = "0.1.1";
  format = "setuptools";

  disabled = pythonOlder "3.7";
```

File: pkgs/development/python-modules/mergedb/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "2034c18dca23456c5b166b63d94300bcd8ec9f386e6cd639c2f66e141c0313f9";
+  };
+
+  nativeBuildInputs = [ git python3Packages.setuptools ];
```
Comment (fabaff) [inline pkgs/development/python-modules/mergedb/default.nix:14]: `git` is not used.

```suggestion
  nativeBuildInputs = [
    setuptools-scm
  ];

```

File: pkgs/development/python-modules/mergedb/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  propagatedBuildInputs = with python3Packages; [
+    setuptools
+    setuptools-scm
+    pyyaml
+    colorama
+    jinja2
+  ];
```
Comment (fabaff) [inline pkgs/development/python-modules/mergedb/default.nix:21]: `setuptools` and its friends are usually not runtime dependencies.

```suggestion
  propagatedBuildInputs = [
    pyyaml
    colorama
    jinja2
  ];

```

File: pkgs/development/python-modules/mergedb/default.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  checkInputs = with python3Packages;  [
+    pytest
+  ];
+  checkPhase = ''
+    pytest
+  '';
```
Comment (fabaff) [inline pkgs/development/python-modules/mergedb/default.nix:27]:
```suggestion
  checkInputs = [
    pytestCheckHook
  ];
```

File: pkgs/development/python-modules/mergedb/default.nix [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+  pythonImportsCheck = [ "mergedb" ];
+
+  meta = with lib; {
+    description = "A tool/library for deep merging YAML files";
+    homepage = "https://github.com/graysonhead/mergedb";
+    license = licenses.gpl3Plus;
```
Comment (fabaff) [inline pkgs/development/python-modules/mergedb/default.nix:34]: There is no mentioning that later releases of the license are supported.

```suggestion
    license = licenses.gpl3Only;
```

File: pkgs/top-level/all-packages.nix [hunk 8]
```diff
@@ -7419,6 +7419,8 @@ with pkgs;
 
   memtester = callPackage ../tools/system/memtester { };
 
+  mergedb = python3Packages.callPackage ../development/python-modules/mergedb { };
```
Comment (fabaff) [inline pkgs/top-level/all-packages.nix:7422]: `mergedb` is a Python module and should be added to `python-packages.nix` instead of `all-packages.nix`.

**Comment by graysonhead** [comment]
@fabaff Thanks for taking a look at this! I believe I've incorporated all of your feedback in my changes.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 167236` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.mergedb</li>
    <li>python39Packages.mergedb</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 167236` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.mergedb</li>
    <li>python39Packages.mergedb</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/167236#issuecomment-1090053519)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
..ew/pr-167236-1]$ ./results/python310Packages.mergedb/bin/mergedb 
usage: mergedb [-h] [--output OUTPUT] [--version] [database_path] [function] [target]

MergeDB is a yaml hierarchical database that allows for customizable merging and inheritance patterns.
[...]
```

Looks good to me.
